### ISSUE TITLE
Failed to set brightness of secondary display - Lenovo ThinkVision L2452p 24-inch

### ISSUE THREAD

**Issue by cafed00d4j** [nriley/brightness#11]
This is the error I get when setting brightness
```
brightness -d 1 0.05
brightness: failed to set brightness of display 0x2b845104 (error -536870201)
```

**Comment by kzahel** [comment]
Same here on Dell U2713H
Wondering if attaching a usb cable to the monitor would help. I don't have one though...

**Comment by cafed00d4j** [comment]
I read somewhere about that trick too (attaching USB cable).. Not really sure how that would work... I did try it out though ... didn't work either

**Comment by chaoli46** [comment]
I find it is almost impossible. You can refer to following materials.
- http://apple.stackexchange.com/a/127258
- http://apple.stackexchange.com/questions/61045/does-apple-support-ddc-ci-for-3rd-party-displays-via-apples-thunderbolt-to-dvi

**Comment by nriley** [comment]
https://github.com/superduper/BrightnessMenulet and/or https://github.com/kfix/ddcctl may work for you.  It sort of works for me with my current display, a LG 27MB85R (contrast adjustment works reliably; brightness doesn't).  See the caveats at the bottom of the ddcctl readme, though; it may also hang your Mac, monitor or both.  (A previous version did indeed hang my Mac.)

In any case, the above seems quite flaky and I don't really want to release software that might crash your Mac or monitor... if Apple decides to support this in any serious way I'll reconsider of course.

**Comment by franklinyu** [comment]
**Really** disappointed at Apple; DDC/CI is so easy on Windows and Linux. Is there anywhere I can yell at Apple about it?

**Comment by auduchinok** [comment]
@franklinyu, try their bug report tracker, bugreport.apple.com. Don't think it'll help though.

**Comment by arcseldon** [comment]
This failed for external Dell P2715QT - 27" Monitor.

```brightness: failed to set brightness of display 0x2b10190e (error -536870201)```

but works for the built-in mac book pro screen.

**Comment by thewellington** [comment]
I have an LG UltraFine 27MD5KA and I get the same error:

    $ brightness -v 1
    brightness: failed to set brightness of display 0x1b561641 (error -536870201)

**Comment by thewellington** [comment]
```
brightness -lv
display 0: main, active, awake, online, external, ID 0x1b561641
	resolution 2560 x 1440 pt (5120 x 2880 px) @ 0.0 Hz, origin (0, 0)
	physical size 599 x 340 mm
	IOKit flags 0x7; IOKit display mode ID 0x80005000
	usable for desktop GUI, uses OpenGL acceleration
```

**Comment by furiousgreencloud** [comment]
brightness -d 0 .5
brightness: failed to set brightness of display 0x7bd8ca3c (error -536870201)

with:

display 0: main, active, awake, online, external, ID 0x7bd8ca3c
	resolution 2560 x 1440 pt (2560 x 1440 px) @ 0.0 Hz, origin (0, 0)
	physical size 597 x 336 mm
	IOKit flags 0x2000007; IOKit display mode ID 0x80005000
	usable for desktop GUI, uses OpenGL acceleration
 
This a Apple iMac Display (Mid 2010)

MacOS 10.10.5 (14F2511)

using 
     https://homebrew.bintray.com/bottles/brightness-1.2.yosemite.bottle.tar.gz

**Comment by jathu** [comment]
This script worked for me: https://github.com/kfix/ddcctl

I'm using an external Acer monitor.

**Comment by latenitefilms** [comment]
MY LG UltraFine 5K also fails when connected to my MacBook Pro via USB-C:

```
brightness -l
display 0: main, active, awake, online, external, ID 0x1b56fe4d
brightness: failed to get brightness of display 0x1b56fe4d (error -536870201)
display 1: active, awake, online, built-in, ID 0x4280c46
display 1: brightness 1.000000
```

However, I can use the brightness keyboard shortcuts to control the brightness of the external monitor, so macOS can definitely do it.

Also, as @jathu suggested, [ddcctl](https://github.com/kfix/ddcctl) also works.

**Comment by thewellington** [comment]
hi @latenitefilms if you download and compile the brightness code yourself it will work with you LG 5K UltraFine display

**Comment by elijahchancey** [comment]
I've got a Mac Mini with an M1 chip, MacOS 11.2.3, and an LG Ultrafine 5k Monitor. Using homebrew, I had a similar error in the issue description. I then downloaded the source, compiled it, put the binary in my path, and it works fine! 

@nriley If you'd like me to provide further information, I'm happy to.

**Comment by nriley** [comment]
Yeah, I think we just need to figure out how to get the Homebrew formula updated... I didn't create the original one!

**Comment by tbrodbeck** [comment]
> hi @latenitefilms if you download and compile the brightness code yourself it will work with you LG 5K UltraFine display

Thanks bud. Same on MacBook Pro 15'' from 2018

**Comment by 016** [comment]
DP's-iMac:~ root# brightness -l
display 0: main, active, awake, online, external, ID 0x78c91d62
brightness: failed to get brightness of display 0x78c91d62 (error -536870201)
display 1: active, awake, online, external, ID 0x78c91cf3
brightness: failed to get brightness of display 0x78c91cf3 (error -536870201)

I got sam error, and I am trying ddcctl and BrightnessMenulet, will update soon.

----- after tested -----
BrightnessMenulet is working for me, just download with the link
https://github.com/superduper/BrightnessMenulet/raw/master/BrightnessMenulet/Brightness_Menulet.zip
unzip, run the app, you will get it in top right corner, click and use it.

I am using 10.11.6 and have 2 ex-monitors it's working.

**Comment by ijoseph** [comment]
> hi @latenitefilms if you download and compile the brightness code yourself it will work with you LG 5K UltraFine display

worked for me, and was very simple to compile too. 

```

╰─ neofetch
                    'c.          <redacted>
                 ,xNMM.          ----------------------------
               .OMMMMo           OS: macOS 12.5.1 21G83 arm64
               OMMM0,            Host: MacBookPro18,3
     .;loddo:' loolloddol;.      Kernel: 21.6.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 3 days, 5 mins
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 241 (brew)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8.1
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 1800x3200, 1512x982
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Dark)
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: tmux
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   CPU: Apple M1 Pro
    kMMMMMMMMMMMMMMMMMMMMMMd     GPU: Apple M1 Pro
     ;KMMMMMMMWXXWMMMMMMMk.      Memory: 3021MiB / 16384MiB
       .cooc,.    .,coo:.
```

**Comment by Alexandro1112** [comment]
same error.

**Comment by foolip** [comment]
I got the same error with my LG UltraFine 27MD5KL and as others on this issue suggested, just building from source and it works. @nriley is the binary in Homebrew outdated?

**Comment by nriley** [comment]
@foolip Yes, it is. I didn't upload the Homebrew package; someone else did and abandoned it.

**Comment by thewellington** [comment]
Just created [PR 49](https://github.com/nriley/brightness/pull/49) to suggest the removal of hombrew from the README if there is no support for updating homebrew.
